oauthenticator/tests/test_globus.py

```diff
@@ -107,9 +103,68 @@ def test_token_exclusion(globus_client, mock_globus_sdk):
         'auth.globus.org'
     ]
     handler = globus_client.handler_for_user(user_model('wash'))
-    user_info = yield authenticator.authenticate(handler)
-    name = user_info['username']
-    assert name == 'wash'
-    auth_state = user_info['auth_state']
-    assert 'globus_data' in auth_state
-    assert list(auth_state['globus_data']['tokens'].keys()) == []
+    data = yield authenticator.authenticate(handler)
+    assert data['name'] == 'wash'
+    assert list(data['auth_state']['tokens'].keys()) == []
+
+
+def test_revoke_tokens(monkeypatch):
+    monkeypatch.setattr(
+        ConfidentialAppAuthClient,
+        'oauth2_revoke_token',
+        Mock()
+    )
+    authenticator = GlobusOAuthenticator()
+    service = {'transfer.api.globus.org': {'access_token': 'foo',
+                                           'refresh_token': 'bar'}}
+    authenticator.revoke_service_tokens(service)
+    assert ConfidentialAppAuthClient.oauth2_revoke_token.called
+
+
+@mark.gen_test
+def test_custom_logout(monkeypatch):
+    custom_logout_url = 'https://universityofindependence.edu/logout'
+    authenticator = GlobusOAuthenticator()
+    logout_handler = mock_handler(GlobusLogoutHandler,
+                                  authenticator=authenticator)
+    monkeypatch.setattr(
+        web.RequestHandler,
+        'redirect',
+        Mock()
+    )
+    logout_handler.clear_login_cookie = Mock()
+    logout_handler.get_current_user = Mock()
+
+    authenticator.logout_redirect_url = custom_logout_url
+    yield logout_handler.get()
+    logout_handler.redirect.assert_called_once_with(custom_logout_url)
+    assert logout_handler.clear_login_cookie.called
+
+
+@mark.gen_test
+def test_logout_revokes_tokens(monkeypatch):
+
+    class User:
+        @gen.coroutine
+        def get_auth_state(self):
+            return {'tokens': {}}
+
+        save_auth_state = Mock()
+        name = 'Wash'
```

Comment:
I went ahead and moved the `User` class here to make it more obvious. For now, this is the only test that should ever need it, due to it being the only one that calls out directly to its methods.